metabox/metabox/core/machine.py

```diff
@@ -94,30 +94,28 @@ def rollback_to(self, savepoint):
         self._container.start(wait=True)
         logger.opt(colors=True).debug(
             "[<y>restored</y>    ] {}", self._container.name)
-        attempt = 0
-        # FIXME: in case containers are restarted after reboot, sometime we hit
-        # the degraded state.
-        # So be sure to wait long enough to claim it started.
-        # https://discuss.linuxcontainers.org/t/snap-lxd-activate-service-hanging-after-system-reboot/8374/16
-        max_attempt = 60
-        old_out = ''
-        while attempt < max_attempt:
-            time.sleep(1)
-            (ret, out, err) = self._container.execute(
-                ['systemctl', 'is-system-running'])
-            if out != old_out:
-                logger.opt(colors=True).debug(
-                    "[<y>{: <12}</y>] {}", out.rstrip(), self._container.name)
-                old_out = out
-            if 'running' in out:
-                break
-            elif 'degraded' in out:
-                break
-            attempt += 1
-        else:
-            raise SystemExit(
-                "Rollback to {} failed (systemd not in running state)".format(
-                    savepoint))
+        if self.config.role == 'service':
+            attempt = 0
+            max_attempt = 60
+            old_out = ''
+            while attempt < max_attempt:
```

Comment:
This can be simplified to:
```
attempts_left = 60
out = ''
while attempts_left and out not in ('starting', 'running', 'degraded'):
   (...)
    attempts_left -= 1
```

Reply:
I've checked the systemctl man pages and indeed there's nothing more than the values above (except a trailing \n). They can be all checked at once.